packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts

```diff
@@ -41,53 +42,70 @@ export const staticResourcesMixin = createSingletonMixin(
     )(superclass);
 
     // the logic should be as following:
-    // if there is a local storage override, use it
-    // otherwise, if there is a base-static-url attribute, use it
-    // otherwise, try to use base-url, and check if it's working
-    // if it's working, use it
-    // if not, use the default content url
+    // 1. if there is a local storage override, use it
+    // 2. otherwise, if there is a base-static-url attribute, use it
+    // 3. otherwise, try to use base-url
+    // 4. otherwise, try to use default content url (BASE_CONTENT_URL)
+    // 5. otherwise, try to use the fallback content url (BASE_CONTENT_URL_FALLBACK)
+    // For steps 3-5: if the url is valid, keep using it, and if not, don't try it anymore
     return class StaticResourcesMixinClass extends BaseClass {
       #lastBaseUrl?: string;
       #workingBaseUrl?: string;
+      #failedBaseUrl = false;
+      #failedContentUrl = false;
 
       #getResourceUrls(filename: string): CustomUrl[] | CustomUrl {
         const overrideUrl = OVERRIDE_CONTENT_URL || this.baseStaticUrl;
 
+        // Steps 1-2: Override URL or base-static-url takes precedence
         if (overrideUrl) {
-          return getResourceUrl({
-            projectId: this.projectId,
-            filename,
-            baseUrl: overrideUrl,
-          });
+          return this.#createResourceUrl(filename, overrideUrl);
```

Comment:
do we need to mark `failedBaseUrl` / `failedContentUrl` as false here?

Reply:
It’s false by default 
we only check when it’s unavailable, not the other way around